src/main/java/com/cloudant/fdblucene/FDBIndexReader.java

```diff
@@ -49,14 +51,43 @@
 
     private static final LeafMetaData LEAF_META_DATA = new LeafMetaData(8, Version.LUCENE_8_1_0, null);
 
-    private TransactionContext txc;
     private final Subspace index;
 
-    public FDBIndexReader(final TransactionContext txc, final Subspace indexSubspace) {
-        this.txc = txc;
+    private final ThreadLocal<Transaction> txnHolder = new ThreadLocal<Transaction>();
+
+    public FDBIndexReader(final Subspace indexSubspace) {
         this.index = indexSubspace;
     }
 
+    public <T> T run(final TransactionContext txc, final IOSupplier<T> retryable) throws IOException {
```

Comment:
ahhh you're enforcing this so that only the calling thread can access the transaction